### PR TITLE
Missile silo jsonify

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
```SUMMARY: Content "Added an improved missile silo."```

#### Purpose of change
Jsonify another hardcoded location.

Hardcoded missile silo was... boring at best. No story, bad layout (sometimes even making some rooms non-accessible), boring loot lying right on the floor and other flaws.

#### Describe the solution
- Basically rewritten the whole location from the scratch;
- Added non-working elevator from the ground floor down to the finale;
- Reduced number of floors from 6 to 5 (I couldn't invent anything for minus 5 floor);
- Slightly reduced the dimensions of the missile;
- Replaced visible nuke gas with invisible radiation;
- -1 floor: meeting room, relax/staff room, bathroom, scientific personnel dorm;
- -2 floor: "science"/computer room, room of (dead) head of the staff, bathroom, two warehouses;
- -3 floor: another, smaller "science" room, (broken) generator room, another warehouse. Also there is a big fault that prevents going the obvious way and forces finding some bypass;
- -4 floor: military staff room, sports room, bathroom, armory;
- silo finale: basically the same layout as it was in hardcoded version, just added some machinery, added shutters to reinforced glass, some other tweaks here and there. Oh, and the sole monster in the whole location - added it mostly as a joke, as it can't pose any threat to player unless he opens the door.

#### Additional context
Ground floor

Minus 1 floor

Minus 2 floor

Minus 3 floor

Minus 4 floor

Finale

### CODE REVIEW

**Comment by ZhilkinSerg** [comment]
FYI:

https://github.com/CleverRaven/Cataclysm-DDA/pull/22366

**Comment by Night-Pryanik** [comment]
Damn, I was sure there was a jsonified versions of hardcoded locations, but I wasn't able to find them. I wonder why we don't move these versions from mod to vanilla for so long?

**Comment by ZhilkinSerg** [comment]
IIrc, there were some balance issues with spawns and also some minor things missing in json definitions (like fields and computers). Anyway, when that would be fixed we will have two variants of jsonized missile silos and that's great.

File: src/computer.cpp [hunk 1]
```diff
@@ -643,18 +643,15 @@ void computer::activate_function( computer_action action )
...
+                if( one_in( 4 ) ) {
+                    g->m.add_field( loc, fd_smoke, rng( 1, 9 ) );
                 }
             }
 
-            explosion_handler::explosion( tripoint( g->u.posx() + 10, g->u.posx() + 21, g->get_levz() ), 200,
-                                          0.7,
-                                          true ); //Only explode once. But make it large.
+            //Only explode once. But make it large.
+            explosion_handler::explosion( nuke_location, 200, 0.7, true );
```
Comment (ZhilkinSerg) [inline src/computer.cpp]:
```suggestion
            explosion_handler::explosion( nuke_location, 200, 0.7, true );
```

**Pull request merged**